frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts

```diff
@@ -669,3 +669,13 @@ export { useModal } from '../app/modal-support/useModal';
  */
 export const ActionServiceProvider: React.FC<ActionServiceProviderProps> = require('@console/shared/src/components/actions/ActionServiceProvider')
   .default;
+
+/**
+ * A component that renders a namespace dropdown menu.
+ * @param {boolean} hideProjects - (optional) A boolean flag that hides project names in the dropdown if set to true. Defaults to false.
+ * @param {React.ReactNode} children - (optional) Elements to render inside the dropdown if hideProjects is set to true
+ * @param {boolean} disabled - (optional) A boolean flag that disables the dropdown if set to true
+ * @param {function} onNamespaceChange - (optional) A function that is executed when a namespace option is selected. The only argument it should accept is new namespace as a string.
+ */
+export const NamespaceBar: React.FC = require('@console/internal/components/namespace')
```

Comment:
We need to add a proper types for the component props (not just the doc)

```ts
type NamespaceBarProps = {
  isDisabled: boolean;
  onNamespaceChange: (namespace: string) => void;
}
```

The `NamespaceBar` component in console/internal should use this type too.


So the `NamespaceBarProps` will go into `sdk/src/api/common-types.ts`

`NamespaceBar` from console internal type will be `NamespaceBarProps & {hideProjects?: boolean}`

Reply:
Done!